### PR TITLE
fix: Resolved Future value ignored warnings from ErrorProne

### LINKED ISSUE #3647: Fix ErrorProne `FutureReturnValueIgnored` warnings
Description: https://errorprone.info/bugpattern/FutureReturnValueIgnored

Dropping exceptions is almost never the right thing to do; 
In order of preference:
* we should be to handle the exceptional flow properly. 
* when there is no reasonable recovery or retry strategy we should at least log the error
* if we are doing this with good cause we should be explicit about it with a comment. 

Taking a quick look at the first occurrence (in the AWS KMS) of this suggests this is not really good-first-start material.

### PR DESCRIPTION
**Pull request by AdityaThakur1998**
### Type of change
- Static-analysis fix + check promotion. **Not behaviour-preserving**: this PR fixes four
  latent hangs, two silent-loss defects, and four tests that could not fail. Those changes
  are isolated to commit 4 and each is covered by a new test.
## Summary
 
Resolves every ErrorProne `[FutureReturnValueIgnored]` warning across the repository, then
promotes the check to ERROR so regressions fail the build.
 
Fixes: https://github.com/kroxylicious/kroxylicious/issues/3647
Supersedes: https://github.com/kroxylicious/kroxylicious/pull/3714 (closed; handed over)
 
`FutureReturnValueIgnored` fires when a method returning `Future`, `CompletableFuture`,
`CompletionStage` or `ChannelFuture` is called as a bare statement. That matters because an
asynchronous failure has no stack to propagate up — it is stored inside the future object, so
if nobody holds that object and nobody asks it whether it failed, the exception is gone
entirely: no log, no stack trace, no test failure. The symptom appears somewhere else, usually
as a timeout.
 
The central decision in this PR is **how each site was classified**. Rather than sorting by
the shape of the call (Netty close vs. write vs. `CompletableFuture` chain), every site was
sorted by a single question: **where does the exception actually end up?** That question
cannot be answered by inspection alone — which is why several sites initially classified as
"safe" moved to "actual bug" once the failure was forced. Where this PR claims an exception is
handled, a test forces the failure and asserts the observable consequence.
 
The result is that **suppression is the minority outcome**. Of the main-source sites, roughly
a third are suppressed; the rest are resolved by logging the failure, by using the value, or
by fixing a defect. Of the ~96 test-source sites, only four are suppressed — in every other
case, using the value makes the test strictly stronger.
 
## Strategy
 
Four categories, corresponding to the four places an exception can end up.
 
**C1 — the exception *is* handled, elsewhere and observably.**
Suppress at the narrowest scope that compiles, with a comment stating the specific reason, and
a test that forces the failure. Five distinct rationales are used, and each site carries the
one that actually applies — not a shared sentence:
 
- *Caller-supplied promise* (`FilterHandler` ×2, `RouteFilterHandler`, `RoutingTerminalHandler`,
  `RouterDispatchHandler`). Netty's documented contract for
  `ChannelOutboundInvoker.write(msg, promise)` is that the promise passed in is notified with
  the outcome. At these sites that promise came from the caller, via an outbound
  `write(ctx, msg, promise)` method, so the caller already holds the object that reports any
  failure. **This is deliberately *not* the argument made in #3714**, which claimed Netty
  returns the same object as the promise — an undocumented implementation detail that was
  rejected in review. The claim here rests only on the promise being notified, and three new
  tests in `FilterHandlerTest` pin it by failing the write at the head of the pipeline and
  asserting the caller's promise completes exceptionally.
- *Void promise* (`FilterHandler`, `KafkaProxyFrontendHandler` ×2, `KafkaProxyBackendHandler`
  ×2, `Action`). `VoidChannelPromise` cannot be listened to by design; failures are delivered
  to `exceptionCaught`. These are hot data paths where void promises exist specifically to
  avoid per-write promise allocation.
- *Terminal `whenComplete` that inspects the throwable* (`EndpointRegistry` ×2,
  `ClientConnectionStateMachine`, `PromiseFactory`, `InternalCompletableFuture`,
  `FilterHandler` ×3, `ApiKeySessionProvider`, `CachingKms`). The failure is handled inside the
  callback; the discarded derived stage can only fail if the callback itself throws.
- *Already logged, retry already scheduled* (`AbstractRefreshingCredentialsProvider` ×2,
  `ApiKeySessionProvider` ×2). On a failed refresh, `propagateResultToFuture` has already
  invoked `onRefreshFailure` — which logs and schedules a backoff retry — *before* the future
  completes. The discarded `thenApply` stage then fails with the same, already-reported cause.
- *Linter false positive* (`VirtualClusterRegistry`). A different sub-check fires on a nested
  `CompletableFuture<CompletableFuture<Void>>`; `thenCompose(Function.identity())` on the
  following line already unwraps it and the composed result is returned.
**C2 — the exception is genuinely lost.** Fixed, not excused. See *Defects found* below.
 
**C3 — the returned value should be used.** `KafkaProxy.close()`, `KafkaClient.close()`,
`MockServer.close()`, `DefaultKroxyliciousTester.spawnProxy`,
`DefaultNetworkBindingOperationProcessor`, `FilterHandler.handleDeferredStage`.
 
**C4 — unrecoverable, nothing to plumb: log it.** 17 `close()` calls in terminal error paths
now attach a shared listener that logs on failure. This is the middle tier named in the issue
("when there is no reasonable recovery or retry strategy we should at least log the error"),
and it resolves the warning outright — **these 17 sites carry no suppression at all**, because
after the change the exception is no longer ignored.
 
**Why not `var ignored = ...`.** Assignment counts as "use" for ErrorProne, so it silences the
check while leaving no searchable marker and no stated reason. It is strictly worse than an
explicit suppression a reviewer can grep for and revisit. The diff contains none; this was
grepped explicitly at the end of the series.
 
**Why method-scope rather than class-scope suppression.** Suppressing at the enclosing method
scopes the exemption to the justified call. With the check promoted to ERROR, a future
future-returning call added elsewhere in the same class still fails compilation and requires
its own reason. Two suppressions (`DefaultNetworkBindingOperationProcessor.start`,
`ImmediateCloseSocketServer`'s constructor) are noted below as slightly broader than ideal.
 
## Defects found
 
The linter was pointing at real bugs. These are the substance of the PR.
 
**Four latent hangs — a caller's future never completed at all.**
`ResilientKms.schedule`, `OauthBearerValidationFilter.schedule` and
`RouterDispatchHandler.executeOnEventLoop` share a shape: build a `CompletableFuture`, schedule
a task that calls a `Supplier`, complete the future from the resulting stage. If the `Supplier`
throws **synchronously** — before returning a stage — the task lambda throws, the exception is
captured by the discarded `ScheduledFuture` (or the event loop's handler), and the caller's
future is **never completed**. Not failed, not cancelled: permanently pending. Each now catches
and propagates. Separately, `RouterDispatchHandler`'s out-of-band response write attached no
promise, so a failed write left `oobFrame.promise()` uncompleted and the OOB caller hanging; it
now completes exceptionally with the write's cause.
 
**`ResponseSequencer` silently lost responses.** Both writes used a default promise, so a
failed response write was dropped **while `nextSequenceToWrite` was still advanced** — the
client loses a response from an ordered stream and the sequencer continues as though ordering
held. Failures are now logged and close the connection (agreed approach, Q1 below).
 
**`KafkaProxy.close()` could return before the proxy was closed.**
`transitionTo(STOPPING, doShutdown, ...)` early-returns *without running* `doShutdown` when the
`STOPPING` transition is already in progress, handing back a `shutdown` future that may still
be incomplete. The old `close()` returned at that point — before `unbindPorts()` had run. This
surfaced as intermittent "Address already in use" failures in `KafkaProxyTest.RunningProxy`. A
controlled run matrix with fixed class ordering showed the failure reproduced with the **old**
`close()` (1 of 3 runs) and not with the new one (0 of 6). Six clean runs is not proof the race
is gone — only that this trigger is — so this is described as no longer observed rather than
fixed.
 
**Four concurrency tests could not fail.** `ActivationTokenTest`'s worker lambdas call
`fail(...)` inside `catch (Exception e)`. `fail()` throws `AssertionError`, which is an `Error`
— so it escapes the catch, is captured by the discarded `Future`, and the `finally` still
counts the latch down. The tests passed regardless of what happened in their worker threads.
Futures are now collected and awaited. Verified by negative control: injecting a failing
worker assertion now produces
`ExecutionException: AssertionFailedError: expected: <99> but was: <1>` and the test fails.
 
## Decisions agreed on the issue
 
Three questions were raised before implementation:
 
| | Question | Agreed |
|---|---|---|
| Q1 | `ResponseSequencer` write failures: void promise, or close the connection? | Close the connection. Tested including the close-while-responses-queued path; the in-flight-response corner case raised did not surface |
| Q2 | `DefaultKroxyliciousTester`: retain the startup future and fail clearly? | Yes — a proxy dying mid-test now fails with an attributable message rather than an unrelated timeout |
| Q3 | Is the promise-notification framing acceptable for the `ctx.write` sites? | Yes |
 
## Changes
 
**69 files, +988 / −192.** 29 main-source files, 39 test-source files, 1 pom.
 
### Commit 1 — `fix(netty): log discarded close() futures` (9 files, +72/−17)
 
Adds `kroxylicious-runtime/.../internal/util/NettyFutures.java`: one shared
`ChannelFutureListener` that logs at WARN with the cause and an operation key-value. Applied at
17 `close()` sites in terminal error paths across `kroxylicious-runtime`,
`kroxylicious-integration-test-support` and `kroxylicious-integration-tests`. No suppressions.
 
Deliberately **not** `ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE`: several of these calls
are inside `exceptionCaught`, where re-firing would re-enter the same handler.
 
### Commit 2 — `fix(proxy): caller-supplied promise writes` (6 files, +101/−5)
 
5 suppressions plus 3 new `FilterHandlerTest` tests proving the caller's promise is notified.
 
### Commit 3 — `fix(proxy): 22 warnings across five rationale groups` (13 files, +171/−1)
 
The C1 groups listed under Strategy. Adds four tests to `FilterHandlerTest` asserting the
connection closes when a filter fails, and one to `PodIdentityCredentialsProviderTest`
asserting a failed fetch is followed by a retry.
 
### Commit 4 — `fix(proxy): latent hangs and silent-loss bugs` (16 files, +381/−38)
 
The defects above, plus C3 plumbing. Nine new tests; every hang fix asserts exceptional
completion **within a bounded timeout**, so a regression fails the test rather than hanging CI.
 
### Commit 5 — `build(errorprone): promote to ERROR` (34 files, +265/−133)
 
The ~96 test-source sites, then the pom argument. Test-source resolutions:
 
| Shape | Resolution |
|---|---|
| `startup()`/`shutdown()` discarded (39) | Captured and asserted. `startup()` returns the future that completes on *shutdown*, so a failed start previously passed silently |
| Teardown shutdowns (20) | `syncUninterruptibly()` — removes a thread/port-leak class of flakiness |
| `EmbeddedChannel` writes (12) | `assertThat(future.cause()).isNull()`. `syncUninterruptibly()` is **not usable** here: `EmbeddedChannel`'s event loop is the test thread and Netty's deadlock guard rejects it. `checkException(promise)` unconditionally calls `setSuccess()` or `setFailure()` before returning, so the future is always complete and the check is tight |
| First `drain()` future (4) | Captured and asserted — those tests are *about* second drain calls |
| `producer.send()` (16) | Collected and awaited, so a rejected produce now fails the test |
| Misc (5) | Captured, or `submit`→`execute` |
 
Only four test-source suppressions: `RoutingDynamicIT` and `RoutingContextContractIT` are
deliberate `acks=0` fire-and-forget tests that already carry explanatory comments — awaiting
those futures would defeat what the tests verify; `ForwardingStyle` schedules in a test plugin
whose executor is shut down in `finally`; `ImmediateCloseSocketServer` keeps `submit()` because
`serverLoop` catches and logs every `Exception` internally and `close()` calls `shutdownNow()`.
 
### Build
 
| File | Change |
|---|---|
| `pom.xml` | Append `-Xep:FutureReturnValueIgnored:ERROR` to the ErrorProne compiler args, following the fix-then-promote pattern of `ReferenceEquality` and `ArrayRecordComponent` |
 
## Testing
 
Unlike a purely annotation-based cleanup, this PR changes runtime behaviour under failure, so
compilation is **not** the verification. New tests:
 
- 7 in `FilterHandlerTest` — caller-promise notification, void-promise routing to
  `exceptionCaught`, and connection close on filter failure (four of these ported from #3714,
  which is where they were originally asked for)
- 2 in `RouterDispatchHandlerTest` — synchronous-throw completing the bridge, and OOB promise
  completion on write failure. The bridge-path test uses a real `DefaultEventExecutorGroup(1)`
  because `EmbeddedEventLoop.inEventLoop()` returns true unconditionally, making the off-event-
  loop path unreachable with `EmbeddedChannel`
- 2 in `ResponseSequencerTest` — close on write failure, including with responses queued
- 2 in `KafkaProxyLifecycleTest` — `close()` waits for shutdown; `close()` twice does not throw
- 1 each in `ResilientKmsTest`, `OauthBearerValidationFilterTest`,
  `PodIdentityCredentialsProviderTest`
- `ActivationTokenTest` ×4 rewritten to await worker futures
Full build green. Integration tests: 1421 run, 0 failures, 1 error —
`ConsumerGroupHeartbeatGroupAuthzIT.shouldEnforceAccessToTopics` ("channel closed before
response received"), which passes in isolation and is in a class this PR does not touch.
Flagged as pre-existing rather than presented as green.
 
Two pre-existing tests needed mock updates from commit 1: `KafkaProxyInitializerTest` and
`KafkaProxyBackendHandlerTest` stub `channel.close()` on Mockito mocks that returned `null`,
which `.addListener(...)` then dereferenced.
 
## Verification
 
```bash
mvn -B clean install -DskipTests 2>/dev/null | grep '\[FutureReturnValueIgnored\]'
# empty, across all modules
mvn -B clean install
# BUILD SUCCESS
```
 
The ratchet was verified to actually engage, since a green build after adding the flag is
indistinguishable from the flag being silently ignored. A discarded future was temporarily
reintroduced in a main-source module (`kroxylicious-runtime`) and a test-source module
(`kroxylicious-integration-tests`); both produced `[ERROR] ... [FutureReturnValueIgnored]` and
`BUILD FAILURE`, then were reverted.
 
The whole-series diff was also grepped for `var ignored` and for any newly introduced local
assigned a `Future`/`CompletionStage`/`ChannelFuture` that is never read — no hits.
 
## Known gaps, flagged rather than hidden
 
- **`AbstractRefreshingCredentialsProvider` test asserts the retry, not the log.** The C1
  rationale says the failure is "already logged"; the test proves the retry half only. The
  provider's tests use WireMock + Awaitility with no log-capture infrastructure, so asserting
  the log would mean introducing that to the module. Happy to add it if wanted.
- **`DefaultKroxyliciousTester` grew more than the other sites** — a record, a
  `createAndStart` helper, a death-cause field, restart-path resets, and close-time surfacing.
  That is a lot of machinery from one discarded future; it follows from the Q2 answer. One
  second `startup()` call remains on the constructor path (idempotent, returns the same future)
  because the `Function<Configuration, AutoCloseable>` factory interface does not expose it;
  this is commented at the call site.
- **Two suppressions are method-scoped over slightly more than the justified call**
  (`DefaultNetworkBindingOperationProcessor.start`, `ImmediateCloseSocketServer`'s
  constructor). Both methods are short, but a future future-returning call added there would be
  silently covered.
- **`executeOnEventLoop` was briefly widened to package-private for testability and has been
  restored to `private`** — the test was made to work without it.
## Suggested follow-up (not in this PR)
 
`KafkaProxyTest.RunningProxy` hardcodes ports 9190/9192/9193 across all its tests while
surefire runs with `<runOrder>random</runOrder>`, so each test depends on the previous one
having fully released them. The `close()` fix above removes one trigger, but the fragility
remains. Happy to raise it as its own issue.
 
---
🤖 Assisted by Claude Code

### CODE REVIEW

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/HaProxyProtocolDetectionHandler.java [hunk 1]
```diff
@@ -94,7 +96,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                                     + "or set proxyProtocol mode to 'allowed' or 'disabled'.");
                     cumulation.release();
                     cumulation = null;
-                    ctx.close();
+                    ctx.close().addListener(logFailure(LOGGER, "close after proxy protocol rejection"));
```
Comment (tombentley) [inline kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/HaProxyProtocolDetectionHandler.java]: "proxy protocol" is pretty ambiguous. In this case the protocol concerned is the [HAProxy `PROXY` protocol](https://www.haproxy.org/download/3.5/doc/proxy-protocol.txt), so making it upper case helps make it less ambiguous. 

```suggestion
                    ctx.close().addListener(logFailure(LOGGER, "close after PROXY protocol rejection"));
```

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java [hunk 2]
```diff
@@ -230,7 +232,7 @@ void inClosed() {
             else {
                 // The outboundChannel can be open without being active during protocol negotiation.
                 // Ensure it gets closed.
-                outboundChannel.close();
+                outboundChannel.close().addListener(logFailure(LOGGER, "close inactive outbound channel"));
```
Comment (tombentley) [inline kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java]: I think we shouldn't use terminology like "outbound". It's not terminology we use elsewhere (or at least, not nearly as much as "upstream" or "proxy-to-broker").

```suggestion
                outboundChannel.close().addListener(logFailure(LOGGER, "close inactive proxy -> broker channel"));
```

File: kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/filter/oauthbearer/OauthBearerValidationFilter.java [hunk 3]
```diff
@@ -247,21 +247,29 @@ private byte[] doAuthenticate(SaslServer server, byte[] authBytes) throws SaslEx
         }
     }
 
-    @SuppressWarnings("java:S1602") // not able to test the scheduled lambda otherwise
+    // FutureReturnValueIgnored: a synchronous throw from operation.get() is now caught and
+    // propagated to `future`, so the scheduled task cannot complete exceptionally and the
+    // discarded ScheduledFuture carries no unobserved failure.
+    @SuppressWarnings({ "java:S1602", "FutureReturnValueIgnored" }) // java:S1602: not able to test the scheduled lambda otherwise
```
Comment (tombentley) [inline kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/filter/oauthbearer/OauthBearerValidationFilter.java]: Let's move the explanation for `// java:S1602` to its own line

File: kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/kms/ResilientKms.java [hunk 4]
```diff
@@ -137,19 +137,30 @@ private static boolean isUnknownEntityException(Throwable e) {
         return e instanceof UnknownAliasException || e instanceof UnknownKeyException;
     }
 
+    // FutureReturnValueIgnored: a synchronous throw from operation.get() is now caught and
```
Comment (tombentley) [inline kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/kms/ResilientKms.java]: the "now" in there won't really make sense once this PR is merged. Let's drop it

File: kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/encryption/RecordEncryptionFilterIT.java [hunk 5]
```diff
@@ -346,10 +346,10 @@ void roundTripManyRecordsFromDifferentProducers(@TopicNameMethodSource Topic top
                 var producer2 = tester.producer();
                 var consumer = tester.consumer()) {
 
-            producer1.send(new ProducerRecord<>(topic.name(), HELLO_WORLD + 1));
-            producer1.send(new ProducerRecord<>(topic.name(), HELLO_WORLD + 2));
+            producer1.send(new ProducerRecord<>(topic.name(), HELLO_WORLD + 1)).get(5, TimeUnit.SECONDS);
+            producer1.send(new ProducerRecord<>(topic.name(), HELLO_WORLD + 2)).get(5, TimeUnit.SECONDS);
```
Comment (tombentley) [inline kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/encryption/RecordEncryptionFilterIT.java]: Given the history of flakiness in many of our ITs, I'm seeing lots of hard coded timeouts for `get()` and wondering if we're going to spend many hours with tests timing out. It would be easy to end up fixing them one by one, finding timeouts that work. I wonder if, instead, factor out a single (or small number, depending on usage) constant for use across all the ITs. Like `DEFAULT_SEND_TIMEOUT`

**Review by tombentley — COMMENTED** [review]
Many thanks @AdityaThakur1998, I really appreciate that you've shared the detailed methodology on this one. The code changes look pretty good to me.

On a practical note, have you observed whether or not this makes a big difference to what's getting logged during the normal operation of the proxy? It might be we were ignoring error previously which end up getting recovered (e.g. via client retries) but are now logged (maybe verbosely) as errors. It think it would be good to check (manually is fine) that we're not emitting a lot of `ERROR` logs where we weren't before for what is normal behaviour.

**Comment by AdityaThakur1998** [comment]
Good call @tombentley  measured it with actual data. Full equal runs, 1,421 tests both sides.

**Zero WARN or ERROR from any code this PR introduced**, on either side.

| | main | branch |
|---|---|---|
| Runtime WARN | 26 | 14 |
| Runtime ERROR | 18 | 10 |
| **From new code** | **0** | **0** |

The difference is just the `TlsIT` port-conflict flake firing on one run and not the other — every line in both came from pre-existing loggers.

Also checked the case I was least sure about: `ResponseSequencer`, where a client disconnecting mid-response makes the write fail and would trigger the new WARN. All six dynamic-routing ITs ran (60 tests) and it produced nothing. Manual testing agreed — five SIGKILLs of a producer mid-send and five garbage-TCP connections killed mid-stream produced only pre-existing decoder WARNs, then a clean close.

Worth noting on levels: the 17 close sites and `ResponseSequencer` all log at WARN. The only new ERROR is `KafkaProxy.close()`, once per lifecycle. So this can't add ERROR noise to the data path even in the worst case.

If `ResponseSequencer` does turn out noisy under real churn, the fix is small — expected peer-gone causes (`ClosedChannelException`, connection reset) at DEBUG, WARN for the rest.

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java (hunk 2 above)
Comment (AdityaThakur1998) [reply kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java]: Fixed it the latest commit

File: kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/kms/ResilientKms.java (hunk 4 above)
Comment (AdityaThakur1998) [reply kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/kms/ResilientKms.java]: Fixed in the latest commit

File: kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/HaProxyProtocolDetectionHandler.java (hunk 1 above)
Comment (AdityaThakur1998) [reply kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/HaProxyProtocolDetectionHandler.java]: Fixed in the latest commit

File: kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/filter/oauthbearer/OauthBearerValidationFilter.java (hunk 3 above)
Comment (AdityaThakur1998) [reply kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/filter/oauthbearer/OauthBearerValidationFilter.java]: Fixed it in the latest commit

File: kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/encryption/RecordEncryptionFilterIT.java (hunk 5 above)
Comment (AdityaThakur1998) [reply kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/encryption/RecordEncryptionFilterIT.java]: Good point. Two constants on `BaseIT`: `DEFAULT_SEND_TIMEOUT_SECONDS` (5s) for producer sends and `DEFAULT_REQUEST_TIMEOUT_SECONDS` (10s) for request futures, since those wait on different things. Every timeout literal this PR introduced now uses one of them — that included two sites in `OffsetFetchGroupAuthzIT`/`OffsetFetchTopicAuthzIT` I'd missed. I also converted the pre-existing `futureA.get(10, …)` in `AbstractFilterIT`, since leaving it as a literal next to the converted line would have been worse than either state.

The values are unchanged — this is purely centralisation, so any future re-tuning is one edit rather than a hunt.

**Review by tombentley — APPROVED** [review]
Thanks @AdityaThakur1998

**Review by robobario — APPROVED** [review]
LGTM, thanks @AdityaThakur1998

**Comment by AdityaThakur1998** [comment]
Correction to my earlier noise numbers 

I ran both sides with `mvn verify -pl kroxylicious-integration-tests`, which rebuilds only that module and resolves `kroxylicious-runtime` from `~/.m2` — which held branch-compiled JARs. Both runs executed against branch runtime code. Since every one of the 17 `logFailure` sites lives in `kroxylicious-runtime`, that made the comparison meaningless.

Re-ran properly (full `mvn clean install` from each worktree root first, contamination verified absent) and counted from the failsafe XML reports on both sides rather than stdout, since that's where the proxy logs actually land.

**The result that matters holds: zero WARN or ERROR from any class this PR touches**, across 1,421 tests and 105 test classes — including all the dynamic-routing ITs that exercise `ResponseSequencer`, which was the site I was least sure about. The branch XML does capture proxy logging (96 lines present from `KafkaRequestDecoder` and `ClientConnectionStateMachine`, both pre-existing and untouched here), so that zero is a real absence rather than a capture artefact.

What I can't give you is a meaningful main-vs-branch WARN count. Surefire runs in random order, so which code paths get exercised varies per run — main happened not to trigger the oversized-frame path at all. A single run per side can't establish a baseline. But it doesn't need to: if nothing new fires on the branch, noise can't have increased.

Manual testing agreed — five SIGKILLs of a producer mid-send and five garbage-TCP connections killed mid-stream produced only pre-existing decoder WARNs, then a clean close.

On levels: the 17 close sites and `ResponseSequencer` log at WARN; the only new ERROR is `KafkaProxy.close()`, once per lifecycle. So this can't add ERROR noise to the data path even in the worst case.
